src/Compilers/Test/Core/TargetFrameworkUtil.cs

```diff
@@ -261,5 +268,37 @@ string getName(MetadataReference m)
         public static IEnumerable<MetadataReference> GetReferencesWithout(TargetFramework targetFramework, params string[] excludeReferenceNames) =>
             GetReferences(targetFramework)
             .Where(x => !(x is PortableExecutableReference pe && excludeReferenceNames.Contains(pe.FilePath)));
+
+        /// <summary>
+        /// Many of our reference assemblies are only used by a subset of compiler unit tests. Having a PackageReference
+        /// to the assemblies here would cause them to be deployed to every unit test we write though. These are non-trivial 
+        /// in size (4+ MB) and we have ~50 test projects so this adds up fast. To keep size down we just add 
+        /// PackageReference on the few projects that and dynamically load here.
+        /// </summary>
+        private static ImmutableArray<PortableExecutableReference> LoadDynamicReferences(string assemblyName, string targetFrameworkName)
+        {
+            if (s_dynamicReferenceMap.TryGetValue(assemblyName, out var references))
+            {
+                return references;
+            }
+
+            try
+            {
+                var name = new AssemblyName(assemblyName);
+                var assembly = Assembly.Load(name);
+
+                var type = assembly.GetType($"Basic.Reference.Assemblies.{targetFrameworkName}", throwOnError: true);
+                var prop = type.GetProperty("All", BindingFlags.Public | BindingFlags.Static);
+                var obj = prop.GetGetMethod()!.Invoke(null, null);
+                references = ((IEnumerable<PortableExecutableReference>)obj).ToImmutableArray();
+                _ = s_dynamicReferenceMap.TryAdd(assemblyName, references);
```

Comment:
nit: instead of discarding, consider capturing the result and asserting on it

Reply:
There isn't anything we can assert though. It's possible for two threads to get here race to add. Who wins though isn't realy that interestig. I more used the `_ =` here to acknowledge the `Try` was happening. I can add a comment to clarify.